app/src/main/java/education/mahmoud/quranyapp/feature/read_log/ReadLogFragment.kt

```diff
@@ -4,40 +4,36 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.Toast
 import androidx.recyclerview.widget.DefaultItemAnimator
 import androidx.recyclerview.widget.ItemTouchHelper
 import androidx.recyclerview.widget.RecyclerView
 import education.mahmoud.quranyapp.R
-import education.mahmoud.quranyapp.base.DataLoadingBaseFragment
+import education.mahmoud.quranyapp.base.BaseFragment
 import education.mahmoud.quranyapp.datalayer.Repository
+import education.mahmoud.quranyapp.utils.show
 import kotlinx.android.synthetic.main.fragment_read_log.*
 import org.koin.android.ext.android.inject
 
 /**
  * A simple [Fragment] subclass.
  */
-class ReadLogFragment : DataLoadingBaseFragment() {
+class ReadLogFragment : BaseFragment(), BaseFragment.InitListener {
 
     private val repository: Repository by inject()
     private var logAdapter: ReadLogAdapter = ReadLogAdapter()
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View? {
-        val view = inflater.inflate(R.layout.fragment_read_log, container, false)
-        return view
+        setOnInitListeners(this)
```

Comment:
here listner is called before view is inflated 
i think this is wrong

Reply:
nope, it is not wrong in this case, since the listener should be registered as early as possible because it will callback the methods from onViewCreated of BaseFragment, and it is NOT something related to callback on a particular view like Button, EditText etc, I hope it makes sense.